chasm/tree.go

```diff
@@ -354,6 +354,41 @@ func (n *Node) prepareDataValue(
 	return nil
 }
 
+func (n *Node) preparePointerValue(
+	chasmContext Context,
+) error {
+	metadata := n.serializedNode.Metadata
+	pointerAttr := metadata.GetPointerAttributes()
+	if pointerAttr == nil {
+		return serviceerror.NewInternal(
+			fmt.Sprintf("expect chasm node to have PointerAttributes, actual attributes: %v", metadata.Attributes),
+		)
+	}
+
+	if n.valueState == valueStateNeedDeserialize {
+		if err := n.deserialize(nil); err != nil {
+			return fmt.Errorf("failed to deserialize data: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (n *Node) findNode(path []string) *Node {
```

Comment:
I added a `getNodeByPath` method some time ago.

Reply:
![](https://user-images.githubusercontent.com/2232524/75076254-832f8f00-54b4-11ea-81b9-1a6648a486a5.png)

Reply:
I am removing mine, but renaming yours to `findNode`. It is a better name.